src/main/java/com/google/devtools/build/lib/sandbox/WindowsSandboxUtil.java

```diff
@@ -0,0 +1,244 @@
+  private static final Logger log = Logger.getLogger(WindowsSandboxUtil.class.getName());
+
+  private static final String WINDOWS_SANDBOX = "BazelSandbox" + OsUtils.executableExtension();
+
+    /**
+   * Checks if the given Windows sandbox binary is available and is valid.
+   *
+   * @param binary path to the Windows sandbox binary that will later be used in the {@link #mount} call.
+   * @return true if the binary looks good, false otherwise
+   * @throws IOException if there is a problem trying to start the subprocess
+   */
+  public static boolean isAvailable(PathFragment binary) {
+    Subprocess process;
+    try {
+      process =
+          new SubprocessBuilder()
+              .setArgv(binary.getPathString(), "-h")
+              .setStdout(StreamAction.STREAM)
+              .redirectErrorStream(true)
+              .setWorkingDirectory(new File("."))
+              .start();
+    } catch (IOException e) {
+      log.warning("Windows sandbox binary at " + binary + " seems to be missing; got error " + e);
+      return false;
+    }
+
+    ByteArrayOutputStream outErrBytes = new ByteArrayOutputStream();
+    try {
+      ByteStreams.copy(process.getInputStream(), outErrBytes);
+    } catch (IOException e) {
+      try {
+        outErrBytes.write(("Failed to read stdout: " + e).getBytes());
+      } catch (IOException e2) {
+        // Should not really have happened. There is nothing we can do.
+      }
+    }
+    String outErr = outErrBytes.toString().replaceFirst("\n$", "");
+
+    int exitCode = waitForProcess(process);
+    if (exitCode == 0) {
+      // TODO(rongjiecomputer): Validate the version number and ensure we support it. Would be nice to reuse
+      // the DottedVersion logic from the Apple rules.
+      return true;
+    } else {
+      log.warning(
+          "Windows sandbox binary at "
+              + binary
+              + " returned non-zero exit code "
+              + exitCode
+              + " and output "
+              + outErr);
+      return false;
+    }
+  }
+
+
+  /**
+   * Waits for a process to terminate.
+   *
+   * @param process the process to wait for
+   * @return the exit code of the terminated process
+   */
+  private static int waitForProcess(Subprocess process) {
+    boolean interrupted = false;
+    try {
+      while (true) {
+        try {
+          process.waitFor();
+          break;
+        } catch (InterruptedException ie) {
+          interrupted = true;
+        }
+      }
+    } finally {
+      if (interrupted) {
+        Thread.currentThread().interrupt();
+      }
+    }
+    return process.exitValue();
+  }
+
+
+  /** Returns a new command line builder for the {@code windows-sandbox} tool. */
+  public static CommandLineBuilder commandLineBuilder(
+      PathFragment windowsSandboxPath, List<String> commandArguments) {
+    return new CommandLineBuilder()
+        .setWindowsSandboxPath(windowsSandboxPath)
+        .setCommandArguments(commandArguments);
```

Comment:
opt: if `windowsSandboxPath` and `commandArguments` are mandatory, then it's safer to set them in the Builder ctor.

Reply:
Done.